server/src/main/java/org/elasticsearch/ingest/SamplingService.java

```diff
@@ -345,11 +396,35 @@ public void deleteSampleConfiguration(
         );
     }
 
+    /*
+     * This version is meant to be used by background processes, not user requests.
+     */
+    private void deleteSampleConfiguration(ProjectId projectId, String index) {
+        deleteSampleConfiguration(projectId, index, TimeValue.MAX_VALUE, TimeValue.MAX_VALUE, ActionListener.wrap(response -> {
+            if (response.isAcknowledged()) {
+                logger.debug("Deleted sampling configuration for {}", index);
+            } else {
+                logger.warn("Deletion of sampling configuration for {} not acknowledged", index);
+            }
+        }, e -> logger.warn("Failed to delete sample configuration for " + index, e)));
+    }
+
     @Override
     public void clusterChanged(ClusterChangedEvent event) {
         if (RANDOM_SAMPLING_FEATURE_FLAG == false) {
             return;
         }
+        final boolean isMaster = event.localNodeMaster();
+        final boolean wasMaster = event.previousState().nodes().isLocalNodeElectedMaster();
+        if (wasMaster != isMaster) {
+            if (isMaster) {
```

Comment:
Does this periodic job _need_ to run on the master? I'd think that this check can run on any node (as long as it only runs on one node), but I obviously don't know the whole sampling feature that well. We could use the persistent task framework just like we do for the health node or the GeoIP downloaders. What do you think?

Reply:
The intention was to keep it simple. Running through 0-19 (most likely 0 or 1) sampling configurations every 30 minutes isn't likely to impact the master enough to be worth the complexity of moving it somewhere else, does it?

Reply:
The (very) slightly heavier work that this spawns is removing the configuration from the cluster state, which does have to run on the master.

Reply:
Right, I keep forgetting that this sampling stuff runs at such a low scale. Makes sense to have it here, thanks for explaining!